vunit/ui/__init__.py

```diff
@@ -708,6 +708,61 @@ def enable_location_preprocessing(
                 preprocessor.remove_subprogram(subprogram)
         self._location_preprocessor = preprocessor
 
+    def add_waves_by_depth(self, depth: int):
+        """
+        Adds all signals in all entities down to the specified depth. By
+        calling this method, it will create a TCL script that adds all
+        signals in all entities and group each entities signals by the
+        path name. Then, it will call `set_sim_option()` with the
+        correct GUI flags.
+
+        :param depth: Hierarchy depth to group and add all signals for \
+            each entity
+
+        :example:
+
+        .. code-block:: python
+
+           prj.add_waves_by_depth(depth=3)
+
+        Which would, for example, add waveforms to the waveform viewer
+        like so:
+        tb/*
+        tb/uut/*
+        tb/uut/ent1/*
+        tb/uut/ent2/*
+
+        .. note::
+           Only affects test benches added *before* the option is set.
```

Comment:
I think it would be desirable to let the option affect different scopes: project, library, specific test... Having it affect "all previously defined tests" feels slightly unidiomatic in the context of VUnit.